chain/client/src/state_request_actor.rs

```diff
@@ -134,21 +138,44 @@ impl StateRequestActor {
             }
         }
     }
+
+    /// Returns the protocol version used in the epoch containing the sync block.
+    /// This is used for synchronization related to the state parts format (see #14013 for details).
+    fn get_protocol_version_from_sync_hash(
+        &self,
+        sync_hash: &CryptoHash,
+    ) -> Option<ProtocolVersion> {
```

Comment:
nit: I think it would be nicer to return an error here and handle it (log error) at the caller side. You can then simplify this method further.